Assets/Plugins/Source/Editor/PreProcessConfigConfirmation.cs

```diff
@@ -32,6 +32,13 @@ class PreProcessConfigConfirmation : IPreprocessBuildWithReport
     public int callbackOrder { get { return 0; } }
     public void OnPreprocessBuild(BuildReport report)
     {
+        // Ensure that we don't do checks for config files if the current platform group has EOS disabled
+        var defineSymbolsForGroup = PlayerSettings.GetScriptingDefineSymbolsForGroup(report.summary.platformGroup);
+        if (defineSymbolsForGroup.Contains("EOS_DISABLE"))
```

Comment:
What's the pros/cons of doing it this way vs adding #if !EOS_DISABLE/endif around the function body? I never thought to check for defines in the player settings itself.

Reply:
One con is that GetScriptingDefineSymbolsForGroup is marked for deprecation.

Reply:
It might be worth making a separate editor assembly that is disabled when EOS_DISABLE is set.

Reply:
I used GetScriptingDefineSymbolsForGroup, instead of GetScriptingDefineSymbols because it's available in all versions in 2020+ so far released, but GetScriptingDefineSymbols, is not. 

The pro is that if something adds a define, using the method to access the defines that will be used in the build will still work, i.e. it works in more cases than just using the EOS_DISABLE conditional compile.

I'm not aware of any particular cons with using the method, except that unity sometimes deprecates methods and adds replacements for them, as in the case that Steven pointed out.

Reply:
For slight future proofing it might be good to use the `PlayerSettings.GetScriptingDefineSymbols` function using a `NamedBuildTarget` received from `UnityEditor.Build.NamedBuildTarget.FromBuildTargetGroup`

Reply:
That said, `UnityEditor.Build.NamedBuildTarget.FromBuildTargetGroup` appears to be undocumented so who know who that'll change.